app/Http/Controllers/JobController.php

```diff
@@ -234,18 +236,30 @@ public function createAsManager(Manager $manager)
         $managerEn = $manager->translate('en');
         $managerFr = $manager->translate('fr');
 
-        $jobPoster->fill([
-            'department_id' => $manager->department_id,
-            'en' => [
-                'branch' => $managerEn->branch,
-                'division' => $managerEn->division,
-            ],
-            'fr' => [
-                'branch' => $managerFr->branch,
-                'division' => $managerFr->division,
-            ]
-        ]);
-        $jobPoster->save();
+        try {
+            $jobPoster->fill([
+                'department_id' => $manager->department_id,
+                'en' => [
+                    'branch' => $managerEn->branch,
+                    'division' => $managerEn->division,
+                ],
+                'fr' => [
+                    'branch' => $managerFr->branch,
+                    'division' => $managerFr->division,
+                ]
+            ]);
+            $jobPoster->save();
+        } catch (\Exception $e) {
+            throw new AdminException(
```

Comment:
It should be possible for these to be left null. Initializing them to the manager's values is just a convenient default.

Reply:
Updated by moving more logic into the twig partial (which I'm not thrilled about, but it seems to be working okay).

Reply:
I didn't realize this would require so much more work in the twig template, I'm sorry 😬

Reply:
Haha no worries, just highlights more refactoring opportunity.